src/negbinfit.jl

```diff
@@ -0,0 +1,81 @@
+function mleForθ(y::AbstractVector, μ::AbstractVector, wts::AbstractVector;
+                 maxIter = 30, convTol = 1.e-6)
+    function firstDeriv(θ::Real)
+        tmp = (y+θ)./(μ+θ) + log.(μ+θ) - 1 - log.(θ) - digamma.(θ+y) + digamma(θ)
```

Comment:
It would be nice to avoid allocating intermediary copies here. You could turn this into an anonymous function to pass to `sum`, or use a generator. Same below and in the main function.